src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -234,35 +238,34 @@ string_exprt string_constraint_generatort::add_axioms_from_int_with_radix(
     // a6 : |res| == size && res[0] is a digit for radix => input_int == sum
     // a7 : |res| == size && res[0] == '-' => input_int == -sum
 
-    exprt::operandst digit_constraints;
-    exprt sum=
-      get_numeric_value_from_character(res[0], char_type, type, radix_ul);
-
-    for(size_t j=1; j<size; j++)
+    if(size>1)
```

Comment:
I've done that. Apologies for leaving out-of-date comments. The main thing to notice is that `sum` and `digit_constraints` persist between iterations of the loop, so we only need to add a little to them to make them correct for the new value of `size`. And you can think of `j` as `size-1` - it's the index of the last character if `|res| == size`.